src/webots/scene_tree/WbSceneTree.cpp

```diff
@@ -1476,16 +1476,38 @@ void WbSceneTree::exportObject() {
     topLevelWidget, tr("Export Webots Object"),
     WbProject::computeBestPathForSaveAs(WbPreferences::instance()->value("Directories/objects").toString() + "/" +
                                         mSelectedItem->node()->modelName() + ".wbo"),
-    tr("Files (*.wbo *.WBO)"));
+    tr("All files(*.wbo *.WBO *.wrl *.WRL *.urdf *.URDF);;Webots object (*.wbo *.WBO);;VRML (*.wrl *.WRL);;URDF (*.urdf "
+       "*.URDF)"));
 
   if (fileName.isEmpty())
     return;
 
-  WbPreferences::instance()->setValue("Directories/objects", QFileInfo(fileName).absolutePath() + "/");
+  const QStringList supportedExtension = QStringList() << ".wbo"
+                                                       << ".wrl"
+                                                       << ".urdf";
+  bool extensionSupported = false;
+  for (int i = 0; i < supportedExtension.size(); ++i) {
+    if (fileName.endsWith(supportedExtension[i], Qt::CaseInsensitive)) {
+      extensionSupported = true;
+      break;
+    }
+  }
+
+  if (!extensionSupported) {
+    WbLog::error(tr("Unsupported '%1' extension.").arg(QFileInfo(fileName).suffix()));
+    return;
+  }
+
+  if (fileName.endsWith("urdf", Qt::CaseInsensitive) && !mSelectedItem->node()->isRobot()) {
+    WbLog::error(tr("URDF exportation available only for Robot nodes."));
+    return;
+  }
 
   QFile file(fileName);
-  if (!file.open(QIODevice::WriteOnly))
+  if (!file.open(QIODevice::WriteOnly)) {
+    WbLog::error(tr("File not writable."));
```

Comment:
Maybe it would be good to also print the path of the file that cannot be written.
Like it is done here:
https://github.com/cyberbotics/webots/blob/66116d3fcd3dc42aa739ff609a0430a73b530f4a/src/webots/gui/WbVideoRecorder.cpp#L553